hypothesis-python/tests/cover/test_settings.py

```diff
@@ -220,6 +242,9 @@ def test_settings_alone():
 """
 
 
+# runpytest_inprocess uses invalidate_caches in pytest, which is not thread safe
+# (I presume; produces keyerrors).
+@skipif_threading
```

Comment:
upstream issue time?  feels like this should be in-scope for quansight

Reply:
I avoided opening any `pytest` issues out of respect for this comment 😅:

> Note that the pytest maintainers have explicitly ruled out making pytest thread-safe, please do not open issues asking to fix thread safety issues in pytest itself.
> https://py-free-threading.github.io/testing/#pytest-is-not-thread-safe

Reply:
Discussion where this probably came from: https://github.com/pytest-dev/pytest/issues/12356#issuecomment-2126686296

If you think upstream would be receptive to an issue here I can open one